.Dislocation Motion.tmp.nlogo

```diff
@@ -240,22 +247,23 @@ to update-lattice-view
   [; lattice-view = hide-atoms
       ask atoms [ hide-turtle ]
   ])
+  set prev-lattice-view lattice-view
 end
 
 to control-temp ; this heats or cools the system based on the average temperature of the system compared to the set system-temp
-  let current-speed-avg mean [ sqrt (vx ^ 2 + vy ^ 2)] of atoms
+  let current-speed-avg mean [ sqrt (vx ^ 2 + vy ^ 2) ] of unpinned-atoms
   let target-speed-avg sqrt-2-kb-over-m * sqrt system-temp
   let scaling-factor target-speed-avg / current-speed-avg
   if current-speed-avg != 0 [
-    ask atoms [
+    ask unpin [
```

Comment:
Doesn't unpin need to be unpinned-atoms?

Reply:
I think what you're commenting on might be a little off because it's the .tmp version. It is unpinned-atoms? in the non-.tmp version